CONTRIBUTING.md

```diff
@@ -71,15 +71,36 @@ git status
 git commit # type a good commit message
 ```
 
-6. Push your commit to GitHub
+6. Bring master up-to-date and rebase
+
+```sh
+git checkout master
+git pull https://github.com/decred/dcrtime.git
+git checkout my-great-stuff
+git rebase -i master
+```
+
+In the text editor, change the command from `pick` to `squash` for **all but the top** commit. If you want to use a different
+commit message for everything, change the command from  `pick` to `reword` on the top commit.
+
+7. **If you have conflicts**, you will need to resolve them by iterating through the diffs one conflicting commit at a time
+
+```sh
+# resolve diffs

```

Comment:
I suggest wording this as `# resolve conflicts`.

And since this is quite confusing and a challenge to those new to git, I think we should link to a good guide or blog post on resolving commits (find the `<<<<` stuff and see how the changes to the same code can/should be combined, etc.